x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/logstash_instructions/index.tsx

```diff
@@ -76,7 +77,7 @@ const CollapsibleCallout: React.FunctionComponent<EuiCallOutProps> = ({ children
           />
         </EuiButton>
       ) : (
-        <EuiButton onClick={() => setIsOpen(true)} fill={true}>
+        <EuiButton onClick={() => setIsOpen(true)} fill={true} disabled={!hasAllSettings}>
```

Comment:
nit: I added a `MissingPrivilegesToolTip` in https://github.com/elastic/kibana/pull/195185 that you can use to display a tooltip on why it's disabled